### PR TITLE
added hibernate-relation-homework solution

### PR DESCRIPTION
**Pull request by Basilius-Third**

### CODE REVIEW

File: src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java [hunk 1]
```diff
@@ -12,11 +15,32 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
...
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(actor);
+            transaction.commit();
+            return actor;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Could not add actor to DB", e);
```
Comment (PashaTerletskyi) [inline src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java:30]:
```suggestion
            throw new DataProcessingException("Could not add actor - " + actor + " to DB", e);
```

File: src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java [hunk 2]
```diff
@@ -12,11 +15,32 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
...
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(country);
+            transaction.commit();
+            return country;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Could not add country to DB", e);
```
Comment (PashaTerletskyi) [inline src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java:30]:
```suggestion
            throw new DataProcessingException("Could not add country - " + country + " DB", e);
```

File: src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java [hunk 3]
```diff
@@ -12,11 +15,32 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
...
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Could not add movie to DB", e);
```
Comment (PashaTerletskyi) [inline src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java:30]:
```suggestion
            throw new DataProcessingException("Could not add movie - " + movie + " to DB", e);
```

**Review by PashaTerletskyi — COMMENTED** [review]
Great!))

**Review by PashaTerletskyi — APPROVED** [review]

File: src/main/java/mate/academy/hibernate/relations/model/Country.java [hunk 4]
```diff
@@ -1,6 +1,16 @@
 package mate.academy.hibernate.relations.model;
 
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "countrys")
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/hibernate/relations/model/Country.java:10]:
```suggestion
@Table(name = "countries")
```

File: src/main/java/mate/academy/hibernate/relations/util/HibernateUtil.java [hunk 5]
```diff
@@ -1,9 +1,20 @@
 package mate.academy.hibernate.relations.util;
 
 import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
 
 public class HibernateUtil {
+    private static SessionFactory instance = initSessionFactory();
+
+    private HibernateUtil() {
+
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/hibernate/relations/util/HibernateUtil.java:10]:
```suggestion
```

File: src/main/java/mate/academy/hibernate/relations/util/HibernateUtil.java [hunk 6]
```diff
@@ -1,9 +1,20 @@
 package mate.academy.hibernate.relations.util;
 
 import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
 
 public class HibernateUtil {
+    private static SessionFactory instance = initSessionFactory();
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/hibernate/relations/util/HibernateUtil.java:7]:
```suggestion
    private static final SessionFactory instance = initSessionFactory();
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**